src/components/HeaderRow/HeaderRow.tsx

```diff
@@ -19,15 +23,27 @@ export interface HeaderRowProps<TData, TValue> {
 }
 
 export const HeaderRow = <TData, TValue>({
-    cellClassName,
-    className,
+    cellClassName: cellClassNameProp,
+    className: classNameProp,
     headerGroup,
     parentHeaderGroup,
     renderResizeHandle,
     renderSortIndicator,
     resizeHandleClassName,
     sortIndicatorClassName,
 }: HeaderRowProps<TData, TValue>) => {
+    const className = React.useMemo(() => {
+        return typeof classNameProp === 'function'
+            ? classNameProp(headerGroup, parentHeaderGroup)
+            : classNameProp;
+    }, [classNameProp, headerGroup, parentHeaderGroup]);
+
+    const cellClassName = React.useMemo(() => {
+        return typeof cellClassNameProp === 'function'
+            ? cellClassNameProp(headerGroup, parentHeaderGroup)
+            : cellClassNameProp;
+    }, [cellClassNameProp, headerGroup, parentHeaderGroup]);
```

Comment:
Why this code is here and not in `HeaderCell.tsx`?
And we should pass `header`/`parentHeader` instead of `headerGroup`/`parentHeaderGroup`